services/src/main/java/io/druid/cli/CliHistorical.java

```diff
@@ -103,9 +114,52 @@ public void configure(Binder binder)
             JsonConfigProvider.bind(binder, "druid.historical.cache", CacheConfig.class);
             binder.install(new CacheModule());
             MetricsModule.register(binder, CacheMonitor.class);
+
+            binder.bind(ForSideEffectsOnlyProvider.Child.class).toProvider(ForSideEffectsOnlyProvider.class).asEagerSingleton();
           }
         },
         new LookupModule()
     );
   }
+
+  private static class ForSideEffectsOnlyProvider implements Provider<ForSideEffectsOnlyProvider.Child>
```

Comment:
refactored it , asEagerSingleton() is not used anymore. Individual CliXXX classes don't have much now.